modules/siddhi-core/src/main/java/org/wso2/siddhi/core/util/parser/StoreQueryParser.java

```diff
@@ -250,6 +391,72 @@ private static void populateFindStoreQueryRuntime(FindStoreQueryRuntime findStor
                 getOutputStreamDefinition().getAttributeList());
     }
 
+    /*
+    * This method is used to pupulate following two types of store query runtimes.
+    *      1. Update Or Insert Store Query Runtime.
+    *      2. Select Insert Store Query Runtime.
+    *      3. Delete Query Runtime.
+    *      4. Update Query Runtime
+    */
+    private static void populateStoreQueryRuntime(StoreQueryRuntime storeQueryRuntime,
+                                                  MatchingMetaInfoHolder matchingMetaInfoHolder,
+                                                  List<VariableExpressionExecutor> variableExpressionExecutors,
+                                                  SiddhiAppContext siddhiAppContext,
+                                                  Map<String, Table> tableMap, Map<String, Window> windowMap,
+                                                  String queryName, int metaPosition, StoreQuery storeQuery,
+                                                  LockWrapper lockWrapper) {
+
+        QuerySelector querySelector = SelectorParser.parse(storeQuery.getSelector(),
+                storeQuery.getOutputStream(), siddhiAppContext, matchingMetaInfoHolder.getMetaStateEvent(),
+                tableMap, variableExpressionExecutors, queryName, metaPosition);
+
+        PassThroughOutputRateLimiter rateLimiter = new PassThroughOutputRateLimiter(queryName);
+        rateLimiter.init(siddhiAppContext, lockWrapper, queryName);
+        OutputCallback outputCallback = OutputParser.constructOutputCallback(storeQuery.getOutputStream(),
+                matchingMetaInfoHolder.getMetaStateEvent().getOutputStreamDefinition(), tableMap, windowMap,
+                siddhiAppContext, true, queryName);
+        rateLimiter.setOutputCallback(outputCallback);
+        querySelector.setNextProcessor(rateLimiter);
+
+        QueryParserHelper.reduceMetaComplexEvent(matchingMetaInfoHolder.getMetaStateEvent());
+        QueryParserHelper.updateVariablePosition(matchingMetaInfoHolder.getMetaStateEvent(),
+                variableExpressionExecutors);
+        querySelector.setEventPopulator(
+                StateEventPopulatorFactory.constructEventPopulator(matchingMetaInfoHolder.getMetaStateEvent()));
+
+        if (storeQueryRuntime instanceof SelectInsertStoreQueryRuntime) {
```

Comment:
Introduce an enum and use switch instead of using if else

Reply:
Fixed with a9c25b981186d8bd5d2454fc331eb07384eeacc7